tests/load/test_read_interfaces.py

```diff
@@ -218,30 +221,48 @@ def double_items():
             except Exception as exc:
                 assert "double_items is not an table" in str(exc)
 
-        # we create a second duckdb pipieline and will see if we can make our filesystem views available there
-        other_pipeline = dlt.pipeline("other_pipeline", dev_mode=True, destination="duckdb")
-        other_db_location = (
-            other_pipeline.destination_client().config.credentials.database  #  type: ignore
-        )
-        other_pipeline.run([1, 2, 3], table_name="items")
-        assert len(other_pipeline._dataset().items.fetchall()) == 3
-
-        # TODO: implement these tests
-        return
+        # we create a duckdb with a table an see wether we can add more views
+        duck_db_location = "_storage/" + uniq_id()
+        external_db = duckdb.connect(duck_db_location)
+        external_db.execute("CREATE SCHEMA first;")
+        external_db.execute("CREATE SCHEMA second;")
+        external_db.execute("CREATE TABLE first.items AS SELECT i FROM range(0, 3) t(i)")
+        assert len(external_db.sql("SELECT * FROM first.items").fetchall()) == 3
 
         # now we can use the filesystemsql client to create the needed views
+        fs_client: Any = pipeline.destination_client()
```

Comment:
this interface is a bit clunky at the moment. we should probably have a unified interface that enables us to mount views into duckdb from any destination that supports it, if you have an idea of where it could live and what the signature should be you can let me know. for the first little while it should be ok, since we are using inner classes here. the secrets are actually also not stored in the database file but in a global directory for duckdb, so I've built it in a way that whoever controls the duckdb connection from the outside can also create secrets with a given key and is then able to delete them again.

Reply:
the interface is already there. it is called `update_stored_schema` on job client. in this case `filesystem` is a staging destination for `duckdb` and you create all tables as views, pass the credentials etc. maybe we need some convenience method that creates `dataset` instance out of such strucutre (so we take not just destination but also staging as input to dataset).

the nice thing is that all this duckdb related code that creates views and does permission handover could go to duckdb client.

https://github.com/dlt-hub/dlt/issues/1692

this is a good followup ticket